examples/twittermap/web/public/javascripts/map/countmap.js

```diff
@@ -69,108 +69,49 @@ angular.module('cloudberry.map')
-      var batchWithoutGeoRequest = cloudberryConfig.querySliceMills > 0 ? {
-        batch: [queryUtil.byTimeRequest(cloudberry.parameters, $scope.geoIdsNotInTimeSeriesCache)],
+      // Handle byGeoTimeRequest query slice option.
+      var byGeoTimeRequestJson = cloudberryConfig.querySliceMills > 0 ? {
+        batch: [queryUtil.byGeoTimeRequest(cloudberry.parameters, $scope.geoIdsNotInTimeSeriesCache)],
         option: {
           sliceMillis: cloudberryConfig.querySliceMills
         }
       } : {
-        batch: [queryUtil.byTimeRequest(cloudberry.parameters, $scope.geoIdsNotInTimeSeriesCache)]
+        batch: [queryUtil.byGeoTimeRequest(cloudberry.parameters, $scope.geoIdsNotInTimeSeriesCache)]
       };
 
       // Gets the Geo IDs that are not in the map result cache.
       $scope.geoIdsNotInCache = MapResultCache.getGeoIdsNotInCache(cloudberry.parameters.keywords,
         cloudberry.parameters.timeInterval,
         cloudberry.parameters.geoIds, cloudberry.parameters.geoLevel);
 
-      // Batch request without time series result - used when the complete time series cache hit, partial map result cache hit case
-      var batchWithoutTimeRequest = cloudberryConfig.querySliceMills > 0 ? {
-        batch: [queryUtil.byGeoRequest(cloudberry.parameters, $scope.geoIdsNotInCache)],
-        option: {
-          sliceMillis: cloudberryConfig.querySliceMills
-        }
-      } : {
-        batch: [queryUtil.byGeoRequest(cloudberry.parameters, $scope.geoIdsNotInCache)]
-      };
-
-      // Batch request with only the geoIds whose map result or time series are not cached yet - partial map result cache hit case
-      // This case also covers the complete cache miss case.
-      var batchWithPartialRequest = cloudberryConfig.querySliceMills > 0 ? {
-        batch: [queryUtil.byTimeRequest(cloudberry.parameters, $scope.geoIdsNotInTimeSeriesCache),
-                queryUtil.byGeoRequest(cloudberry.parameters, $scope.geoIdsNotInCache)],
-        option: {
-          sliceMillis: cloudberryConfig.querySliceMills
+      // Complete time series cache hit case - exclude time series request
+      if($scope.geoIdsNotInTimeSeriesCache.length === 0) {
+        cloudberry.countmapPartialMapResult = MapResultCache.getValues(cloudberry.parameters.geoIds, cloudberry.parameters.geoLevel);
+
+        var emptyStore = new HashMap();
+        if (cloudberry.parameters.geoLevel === "city") {
+          var newCities = [];
+          for (i = 0; i < cloudberry.parameters.geoIds.length; i++) {
+                newCities.push(cloudberry.parameters.geoIds[i]);
+          }
+          PopulationCache.loadCityPopulationToCache(newCities).done(function(){
+            cloudberry.countmapPartialMapResult = MapResultCache.getValues(cloudberry.parameters.geoIds, cloudberry.parameters.geoLevel);
+            cloudberry.countmapMapResult = PopulationCache.getCountMapValues($scope.geoIdsNotInCache, cloudberry.parameters.geoLevel,
```

Comment:
For this reason and according to my understanding of this new design, my suggestion is that: All `countmap` results will come from `TimeSeries` module, all `TimeSeries` module queries will be handled by `TimeSeries/controllers.js-TimeSeriesCtrl` and once it gets some update on `TimeSeries` results, it notifies the `Countmap.js-countMapCtrl` to update its results. Once `Countmap.js-countMapCtrl` gets message saying results should be updated, it will call `TimeSeriesCache` module's API function to get the `{state/county/city, count}` part data, and call `PopulationCache` module's API function to add `population` attribute to the data and accumulate to the final `{state/county/city, count, population}` data format.

Then I think the logic of this part will look more clear and easy to read.